### ISSUE TITLE
swiotlb buffer is full

### ISSUE THREAD

**Issue by dungcules** [google-coral/edgetpu#408]
I am working with coral board and a camera SDK. I am able to run a few examples with this camera SDK. However, one examples returns me the a full swiotlb buffer. In dmesg, following errors are reports
`[  437.548691] dwc3 38200000.dwc3: swiotlb buffer is full (sz: 16384 bytes)`
`[  437.555402] dwc3 38200000.dwc3: DMA: Out of SW-IOMMU space for 16384 bytes ` 
`[  437.562892] usb 2-1.2: usbfs: usb_submit_urb returned -11`

To increase this buffer size, I added `swiotlb=65536` into Kernel command line by modify boot.txt file and then build boot.src. My current command line is as following(PARTUUID value is removed to upload):

`[    0.000000] Kernel command line: console=ttymxc0,115200 console=tty0 earlycon=ec_imx6q,0x30860000,115200 root=PARTUUID= rootfstype=ext4 rw rootwait init=/sbin/init net.ifnames=0 pci=pcie_bus_perf iommu=soft swiotlb=65536
`

The same error messages are report and the the buffer size remains 16384 bytes.

I need help to change this buffer size but my search does not help at all. If you have any idea to change this buffer, please let me know. 

Thanks

**Comment by manoj7410** [comment]
@dungcules Can you try the following method:

=> setenv othbootargs swiotlb=65536
=> printenv othbootargs
othbootargs=swiotlb=65536
=> saveenv

**Comment by dungcules** [comment]
> @dungcules Can you try the following method:
> 
> => setenv othbootargs swiotlb=65536
> => printenv othbootargs
> othbootargs=swiotlb=65536
> => saveenv

The setting of my environment is below:
`u-boot=> printenv`
`baudrate=115200`
`bootcmd=ext2load mmc 0:1 ${loadaddr} boot.scr; source; boota mmc0 boot_a`
`bootdelay=10`
`bootdev=0`
`ethaddr=`
`ethprime=FEC`
`fastboot.mmc_size=15678308352`
`fastboot_dev=mmc0`
`fdt_high=0xffffffffffffffff`
`fdtcontroladdr=ff931390`
`initrd_high=0xffffffffffffffff`
`loadaddr=0x40480000`
`m4boot=ext2load mmc ${bootdev}:1 ${m4loadaddr} ${m4image}; dcache flush; bootaux ${m4loadaddr}`
`m4image=m4_fw.bin`
`m4loadaddr=0x007E0000`
`othbootargs=swiotlb=65536`
`serial#=`
`soc_type=imx8mq`
`splashpos=m,m`
`stderr=serial`
`stdin=serial`
`stdout=serial
`

I am using 4GB Coral Development board. The same error messages with 16384 bytes buffer still remain. Is that any method that help me to retrieve the current size of this swiotlb buffer and its settings?

**Comment by manoj7410** [comment]
Can you paste the output of the below command after reboot:

dmesg | grep command

**Comment by dungcules** [comment]
This is the result of the command after reboot:

`# dmesg | grep command`
`[    0.000000] Kernel command line: console=ttymxc0,115200 console=tty0 earlycon=ec_imx6q,0x30860000,115200 root=PARTUUID=70672ec3-5eee-49ff-b3b1-eb1fbd406bf5 rootfstype=ext4 rw rootwait init=/sbin/init net.ifnames=0 pci=pcie_bus_perf iommu=soft swiotlb=65536`
`#
`

**Comment by manoj7410** [comment]
Are there any particular steps that you can share to reproduce this error ?

**Comment by dungcules** [comment]
I think I found the root cause of this. The issue does not come from the board itself but from the SDK that I used. I am working with a camera SDK and the code is built when the size of buffer is 16KB. After I update this buffer size to 64KB without rebuilt the SDK correctly, the old buffer size is used for these builds. There are some other issues with matching clock between my board and the one generate the source code. Then, my build does not update anything (using MakeFile). I will work with the SDK developer with this issue. However, it would be nice if there is a way to confirm the size of this buffer instead of using the dmesg or uboot for embedded systems which will be deployed in a closed and offline device.
